routes/api/contacts.js

```diff
@@ -1,25 +1,151 @@
-})
+    const results = array.slice(startIndex, endIndex);
+    return results;
+  };
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  contactsRouter.get('/', async (req, res, next) => {
+    const { id: userId } = req.user;
+    try {
+      const contacts = await listContacts(userId);
+      const { page = 1, limit = 20, favorite } = req.query;
 
-module.exports = router
+      let filteredContacts = contacts;
+
+      if (favorite === 'true') {
+        filteredContacts = contacts.filter(contact => contact.favorite);
+      }
+      const paginatedContacts = paginatedResults(filteredContacts, page, limit);
+
+      return res.status(200).json({
+        status: 'success',
+        code: 200,
+        data: { contacts: paginatedContacts },
+      });
+    } catch (error) {
+      res.status(500).json({
+        status: 'error',
+        code: 500,
+        message: `An error occurred while getting the contact list: ${err}`,
+      });
+    }
+  });
+
+  contactsRouter.get('/:id', checkContact, async (req, res, next) => {
+    const { id: userId } = req.user;
+    const { id: contactId } = req.params;
+
+    try {
+      const contact = await getContactById(userId, contactId);
+
+      return res.status(200).json({
+        status: 'success',
+        code: 200,
+        data: { contact },
+      });
+    } catch (err) {
+      res.status(500).json(`An error occurred while getting the contact: ${err}`);
+    }
+  });
+
+  contactsRouter.post('/', async (req, res, next) => {
+    const body = req.body;
+    const { id: userId } = req.user;
+
+    if (Object.keys(body).length === 0) {
+      return res.status(400).json('Error! Missing fields! Empty request is not allowed');
+    }
+
+    try {
+      const contact = await addContact(body, userId);
+
+      return res.status(201).json({
+        status: 'success',
+        code: 201,
+        data: { contact },
+      });
+    } catch (err) {
+      res.status(500).json(`An error occurred while adding the contact: ${err}`);
+    }
+  });
+
+  contactsRouter.delete('/:id', checkContact, async (req, res, next) => {
+    const { id: userId } = req.user;
+    const { id: contactId } = req.params;
+
+    try {
+      await removeContact(contactId, userId);
+
+      return res.status(200).json({
+        message: `Contact with ID ${contactId} has been successfully removed.`,
+      });
+    } catch (err) {
+      res.status(500).json(`An error occurred while removing the contact: ${err}`);
+    }
+  });
+
+  contactsRouter.put('/:id', checkContact, async (req, res, next) => {
+    const { id: userId } = req.user;
+    const { id: contactId } = req.params;
+    const { body } = req;
+
+    if (Object.keys(body).length === 0) {
+      return res.status(400).json('Error! Missing fields! Empty request is not allowed');
+    }
+
+    try {
+      const updatedContact = await updateContact(contactId, body, userId);
+
+      return res.json({
```

Comment:
please add status even if you know that default is `200`. its easier to understand :)